libs/s25main/factories/BuildingFactory.cpp

```diff
@@ -43,8 +44,19 @@ noBuilding* BuildingFactory::CreateBuilding(GameWorldBase& gwg, const BuildingTy
         default: bld = new nobUsual(type, pt, player, nation); break;
     }
     gwg.SetNO(pt, bld);
+    GamePlayer &playerObject = gwg.GetPlayer(player);
     // Don't do this in ctor as building might not be fully initialized yet
-    gwg.GetPlayer(player).AddBuilding(bld, type);
+    playerObject.AddBuilding(bld, type);
+    const auto &description = BLD_WORK_DESC[type];
+    // We're filtering out JOB_PRIVATE because military buildings do their own
+    // worker stuff.
+    // We also can't do this in the nobUsual constructor, because at the point
```

Comment:
Remove as already covered by line 46/48

Reply:
I’d like to replace the comment in line 46/48 then, because I think my comment is more useful. What do you think?